### ISSUE TITLE
Can't get the widget

### ISSUE THREAD

**Issue by Nairolf76** [Shingyx/BoomSwitch#21]
OnePlus 12 & Oxygen OS 14.

The app works fine but I can't find how to add a widget.
Ideally I'd like to be able to automate the switch on with Google Assistant 🙏.

Thanks for the great work.

**Comment by Shingyx** [comment]
Hey there, thanks for reporting.

For the widget issue, can you please let me know what launcher you are using? If you are using the stock OnePlus launcher, do you still see the issue when using a third party launcher, such as [Nova Launcher](https://play.google.com/store/apps/details?id=com.teslacoilsw.launcher)?

Regarding Google Assistant, I don't have plans to implement this feature in the near future, but might reconsider if there's more interest. In the meantime, it might be possible with a separate task automation app but I've personally never tried them. Sorry about this.

**Comment by Nairolf76** [comment]
You are right, I'll using the native OnePlus launcher. The widget is available if I switch to Nova. Would you know how to make it available for 1+ launcher as well, please?

Ok for the assistant 😇

**Comment by Shingyx** [comment]
Unfortunately, this looks like a bug with the OnePlus launcher. Other OnePlus users are running into the same issue since OxygenOS 12 for other apps. https://community.oneplus.com/thread/1534009

I've been experimenting with an idea of creating the shortcut/widget from within the app. I think this may work, but unfortunately this adds a "badge" onto the icon, which I don't think looks very good.

![image](https://github.com/Shingyx/BoomSwitch/assets/11037113/d2cf4548-4dbd-4510-85c6-78f59408bfa1)

Assuming you'd prefer to continue using your current launcher, I think a good workaround might be to use an app like [Shortcut Maker](https://play.google.com/store/apps/details?id=rk.android.app.shortcutmaker):

1. Open Shortcut Maker
2. Tap the "In-App" option
3. Select "BOOM Switch"
4. A pop-up should appear - select your speaker
5. Click "Create Shortcut"
6. Drag the shortcut icon onto your launcher

Can you please give this workaround a try? Thanks.

**Comment by Shingyx** [comment]
I've added a way to create a pinned shortcut from the app's menu on the top right. This creates a shortcut for the selected speaker.

I believe this change should fix this issue. I'll release this shortly.

<img width="420" height="1107" alt="Image" src="https://github.com/user-attachments/assets/e6eb35b2-0267-498a-9832-5897cccaf97e" />
